src/get-started/flutter-for/ios-devs.md

```diff
@@ -5,467 +5,838 @@ description: Learn how to apply iOS developer knowledge when building Flutter ap
-which form your layout.
+<div class="tab-pane active" id="swiftui" role="tabpanel" aria-labelledby="swiftui-tab" markdown="1">
 
-In Flutter, the rough equivalent to a `UIView` is a `Widget`.
-Widgets don't map exactly to iOS views,
-but while you're getting acquainted with how Flutter works
-you can think of them as "the way you declare and construct UI".
 
-However, these have a few differences to a `UIView`.
-To start, widgets have a different lifespan: they are immutable
-and only exist until they need to be changed.
-Whenever widgets or their state change,
-Flutter’s framework creates a new tree of widget instances.
-In comparison, an iOS view is not recreated when it changes,
-but rather it's a mutable entity that is drawn once
-and doesn't redraw until it is invalidated using `setNeedsDisplay()`.
+## SwiftUI Overview
 
-Furthermore, unlike `UIView`, Flutter’s widgets are lightweight,
-in part due to their immutability.
-Because they aren't views themselves,
-and aren't directly drawing anything,
-but rather are a description of the UI and its semantics
-that get "inflated" into actual view objects under the hood.
+This section gives an overview of the core similarities 
+and differences between SwiftUI and Flutter.
 
-Flutter includes the [Material Components][] library.
-These are widgets that implement the
-[Material Design guidelines][].
-Material Design is a flexible design system
-[optimized for all platforms][], including iOS.
+### Views vs. Widgets
 
-But Flutter is flexible and expressive enough
-to implement any design language.
-On iOS, you can use the [Cupertino widgets][]
-to produce an interface that looks like
-[Apple's iOS design language][].
+Flutter and SwiftUI are both declarative frameworks, 
+meaning the code simply describes how the UI should look 
+and what it should do. 
 
-### How do I update widgets?
+In SwiftUI, you use **_views_** 
+to represent parts of your app's UI, 
+and you can provide modifiers that are used to configure those views. 
 
-To update your views on iOS, you directly mutate them.
-In Flutter, widgets are immutable and not updated directly.
-Instead, you have to manipulate the widget’s state.
+<img src="/assets/images/docs/get-started/ios/swiftui-modifiers.png" alt="SwiftUI Text view uses a padding modifier">
 
-This is where the concept of Stateful vs Stateless widgets
-comes in. A `StatelessWidget` is just what it sounds
-like&mdash;a widget with no state attached.
+Flutter is similar in that it uses **_widgets_** 
+to represent parts of your app's UI. Both views and widgets are immutable, 
+meaning they only exist until they need to be changed. 
+Additionally, both SwiftUI and Flutter are designed to use composition 
+rather than inheritance. However, Flutter uses individual widget classes 
+for many aspects of the UI. This means that properties 
+that might normally be represented as modifiers in SwiftUI 
+are instead represented as separate widgets in Flutter. 
```

Comment:
What is "relate classes"?

This doesn't make sense to me because you're saying they're the same and then you say in practice they're different

Reply:
Ah, fair. The explanation I found says composition and inheritance both handle how classes relate to each other. Inheritance works as each subclass is a version of its parent class; composition works through referring to its parent class. Inheritance breaks when the parent class changes; composition doesn't.

Reply:
I wrote the following: 

```
Flutter represents UI components as _widgets_. 
Both views and widgets only exist until they need to be changed.
These languages call this property _immutability_.
Additionally, SwiftUI and Flutter are relate classes using composition, 
rather than inheritance. 
However, Flutter uses individual widget classes 
for many aspects of the UI. This means that properties 
that might normally be represented as modifiers in SwiftUI 
are instead represented as separate widgets in Flutter. 
```